wled00/cfg.cpp

```diff
@@ -694,37 +694,71 @@ bool deserializeConfig(JsonObject doc, bool fromFS) {
   CJSON(macroCountdown, cntdwn["macro"]);
   setCountdown();
 
+  // Load timers into new vector-based system
+  clearTimers();
+  
+  // Load regular timers from "ins" array
   JsonArray timers = tm["ins"];
-  uint8_t it = 0;
   for (JsonObject timer : timers) {
-    if (it > 9) break;
-    if (it<8 && timer[F("hour")]==255) it=8;  // hour==255 -> sunrise/sunset
-    CJSON(timerHours[it], timer[F("hour")]);
-    CJSON(timerMinutes[it], timer["min"]);
-    CJSON(timerMacro[it], timer["macro"]);
-
-    byte dowPrev = timerWeekday[it];
-    //note: act is currently only 0 or 1.
-    //the reason we are not using bool is that the on-disk type in 0.11.0 was already int
-    int actPrev = timerWeekday[it] & 0x01;
-    CJSON(timerWeekday[it], timer[F("dow")]);
-    if (timerWeekday[it] != dowPrev) { //present in JSON
-      timerWeekday[it] <<= 1; //add active bit
-      int act = timer["en"] | actPrev;
-      if (act) timerWeekday[it]++;
+    // Extract timer data from JSON
+    uint8_t hour = timer[F("hour")] | 0;
+    int8_t minute = timer["min"] | 0;
+    uint8_t preset = timer["macro"] | 0;
+    
+    // Handle weekdays and enabled state
+    uint8_t weekdays = (timer[F("dow")] | 0) << 1; // shift weekdays to upper 7 bits
+    if (timer["en"] | false) weekdays |= 0x01; // set enabled bit if timer is enabled
+    
+    // Handle date range for regular timers
+    JsonObject start = timer["start"];
+    JsonObject end = timer["end"];
+    uint8_t monthStart = start["mon"] | 1;
+    uint8_t dayStart = start["day"] | 1;
+    uint8_t monthEnd = end["mon"] | 12;
+    uint8_t dayEnd = end["day"] | 31;
+    
+    // Add regular timer to vector system
+    addTimer(preset, hour, minute, weekdays, monthStart, monthEnd, dayStart, dayEnd);
+  }
+  
+  // Load sunrise/sunset timers from separate "sunrise_sunset" object
+  JsonObject sunriseSunset = tm["sunrise_sunset"];
```

Comment:
No need to have such long key, as this is not a UI element. Use short keys (not necessarily descriptive) like "ss".

Also. Why would you limit yourself to just one "sunrise" and "sunset" event?
You should be able to schedule different sunrise and sunset events as well.